src/pvi/_format/screen.py

```diff
@@ -417,18 +417,30 @@ def generate_component_formatters(
                     c.children
                 ), "Header length does not match number of elements"
 
+                if add_label:
+                    # Scale the headers by the width with the label then
+                    # indent them
+                    original_width = len(c.layout.header) * (
+                        component_bounds.w + self.layout.spacing
+                    )
+                    label_width = self.layout.label_width + self.layout.spacing
+
+                    scaling_factor = (original_width - label_width) / original_width
+                    component_bounds.w = int(component_bounds.w * scaling_factor)
+
+                    component_bounds.indent(label_width)
+
                 # Create column headers
                 for column_header in c.layout.header:
                     yield self.widget_formatter_factory.header_formatter_cls(
                         component_bounds.copy(), column_header
                     )
                     component_bounds.x += component_bounds.w + self.layout.spacing
 
-                # Reset x and shift down y
-                component_bounds.x = bounds.x
+                # Reset x and w and shift down y
+                component_bounds = bounds.copy()
```

Comment:
And then here just do the shift down
```suggestion
                # Shift rows below inserted header row
```

Reply:
[Context](https://github.com/epics-containers/pvi/pull/63#discussion_r1401992222)